crates/store/src/db/models/queries/transactions.rs

```diff
@@ -126,14 +173,191 @@ pub struct TransactionSummaryRowInsert {
+        //
+        // Note: 500 bytes per output note is an over-estimate but ensures we don't
+        // exceed memory limits when these transactions are later converted to proto records.
+        let header_base_size = 4 + 32 + 16 + 64; // block_num + tx_id + account_id + commitments
+        let input_notes_size = transaction_header.input_notes().len() * 32;
+        let output_notes_size = transaction_header.output_notes().len() * 500;
+        let size_in_bytes = (header_base_size + input_notes_size + output_notes_size) as i64;
+
         Self {
-            transaction_id: transaction_id.to_bytes(),
-            account_id: account_id.to_bytes(),
+            transaction_id: transaction_header.id().to_bytes(),
+            account_id: transaction_header.account_id().to_bytes(),
             block_num: block_num.to_raw_sql(),
+            initial_state_commitment: transaction_header.initial_state_commitment().to_bytes(),
+            final_state_commitment: transaction_header.final_state_commitment().to_bytes(),
+            input_notes: input_notes_binary,
+            output_notes: output_notes_binary,
+            size_in_bytes,
+        }
+    }
+}
+
+/// Select complete transaction records for the given accounts and block range.
+///
+/// # Parameters
+/// * `account_ids`: List of account IDs to filter by
+///     - Limit: 0 <= size <= 1000
+/// * `block_range`: Range of blocks to include inclusive
+///
+/// # Returns
+/// A tuple of (`last_block_included`, `transaction_records`) where:
+/// - `last_block_included`: The highest block number included in the response
+/// - `transaction_records`: Vector of transaction records, limited by payload size
+///
+/// # Note
+/// This function returns complete transaction record information including state commitments
+/// and note IDs, allowing for direct conversion to proto `TransactionRecord` without loading
+/// full block data. We use a chunked loading strategy to prevent memory exhaustion attacks and
+/// ensure predictable resource usage.
+///
+/// # Raw SQL
+/// ```sql
+/// SELECT
+///     account_id,
+///     block_num,
+///     transaction_id,
+///     initial_state_commitment,
+///     final_state_commitment,
+///     input_notes,
+///     output_notes,
+///     size_in_bytes
+/// FROM transactions
+/// WHERE block_num > ?1 AND block_num <= ?2 AND account_id IN rarray(?3)
+/// ORDER BY block_num ASC
+/// LIMIT ?4 OFFSET ?5
+/// ```
+/// Note: The query is executed in chunks of 1000 transactions to prevent loading
+/// excessive data and to stop as soon as the accumulated size approaches the 5MB limit.
+pub fn select_transactions_records(
+    conn: &mut SqliteConnection,
+    account_ids: &[AccountId],
+    block_range: RangeInclusive<BlockNumber>,
+) -> Result<(BlockNumber, Vec<crate::db::TransactionRecord>), DatabaseError> {
+    const MAX_PAYLOAD_BYTES: usize = 5 * 1024 * 1024; // 5 MB
+    const CHUNK_SIZE: i64 = 1000; // Read 1000 transactions at a time
```

Comment:
I would maybe mention that given the size of the note records, 1000 records are guaranteed never to return more than about 60MB of data.